src/cnfizers/Tseitin.cc

```diff
@@ -1,357 +1,160 @@
-        else if (!logic.isNot(ptr) && sz > 0) {
+        else if (!logic.isNot(ptr) && sz > 0) { // do not recurse into atoms
             goto tseitin_end;
         }
-        {
-            Pterm const& pt = logic.getPterm(ptr);
-            for (int i = 0; i < pt.size(); i++) {
-                unprocessed_terms.push(pt[i]); // Using the PTRef is safe if a reallocation happened
-            }
+        for (PTRef child : logic.getPterm(ptr)) {
+            unprocessed_terms.push(child);
         }
 tseitin_end:
-        alreadyCnfized.insert(ptr, current_frame_term);
+        alreadyCnfized.insert(ptr, currentFrameId);
     }
-
-    return res;
 }
 
-
-bool Tseitin::cnfizeAnd(PTRef and_term)
-{
-//  assert( list );
-//  assert( list->isList( ) );
-    //
+void Tseitin::cnfizeAnd(PTRef and_term) {
     // ( a_0 & ... & a_{n-1} )
-    //
     // <=>
-    //
     // aux = ( -aux | a_0 ) & ... & ( -aux | a_{n-1} ) & ( aux & -a_0 & ... & -a_{n-1} )
-    //
     Lit v = this->getOrCreateLiteralFor(and_term);
-    vec<Lit> little_clause;
     vec<Lit> big_clause;
     int size = logic.getPterm(and_term).size();
     big_clause.capacity(size + 1);
-    little_clause.push(~v);
-    big_clause   .push(v);
-    bool res = true;
-    for (int i = 0; i < size; i++) {
+    big_clause.push(v);
+    for (int i = 0; i < size; ++i) {
         PTRef arg = logic.getPterm(and_term)[i];
-        little_clause.push( this->getOrCreateLiteralFor(arg) );
-        big_clause   .push(~this->getOrCreateLiteralFor(arg));
-        res &= addClause(little_clause);        // Adds a little clause to the solver
-        little_clause.pop();
+        Lit argLit = this->getOrCreateLiteralFor(arg);
+        big_clause.push(~argLit);
+        addClause({~v, argLit});
     }
-    res &= addClause(big_clause);                    // Adds a big clause to the solver
-    return res;
+    addClause(std::move(big_clause));
 }
 
-
-
-bool Tseitin::cnfizeOr(PTRef or_term)
-{
-    //
+void Tseitin::cnfizeOr(PTRef or_term) {
     // ( a_0 | ... | a_{n-1} )
-    //
     // <=>
-    //
     // aux = ( aux | -a_0 ) & ... & ( aux | -a_{n-1} ) & ( -aux | a_0 | ... | a_{n-1} )
-    //
-    vec<Lit>    little_clause;
-    vec<Lit>    big_clause;
     Lit v = this->getOrCreateLiteralFor(or_term);
-    little_clause.push( v);
-    big_clause   .push(~v);
-    bool res = true;
-    for (int i = 0 ; i < logic.getPterm(or_term).size(); i++) {
-        Lit arg = this->getOrCreateLiteralFor(logic.getPterm(or_term)[i]);
-        little_clause.push(~arg);
-        big_clause   .push( arg);
-
-        res &= addClause(little_clause);        // Adds a little clause to the solver
-
-        little_clause.pop();
+    vec<Lit> big_clause;
+    int size = logic.getPterm(or_term).size();
+    big_clause.push(~v);
+    for (int i = 0; i < size; ++i) {
+        PTRef arg = logic.getPterm(or_term)[i];
+        Lit argLit = this->getOrCreateLiteralFor(arg);
+        big_clause.push(argLit);
```

Comment:
Maybe add Tseitin to the clang-format, almost whole file is changed anyway(and Cnfizer, MainSolver, etc.)